addons/pkg-npm/common/src/main/java/org/commonjava/indy/pkg/npm/content/PackageMetadataGenerator.java

```diff
@@ -321,11 +321,18 @@ private boolean writePackageMetadata( List<StoreResource> firstLevelFiles, Artif
                 }
 
                 // Generate tarball url if missing
+                String tarball = "http://indy/" + packagePath.getTarPath(); // here we use mock host. indy will amend it with the right hostname
                 if ( versionMetadata.getDist() == null )
                 {
-                    String tarball = "http://indy/" + packagePath.getTarPath(); // here we use mock host. indy will amend it with the right hostname
                     //logger.debug( "Generate dist tarball: {}", tarball );
-                    versionMetadata.setDist( new Dist(  tarball ) );
+                    versionMetadata.setDist( new Dist( tarball ) );
+                } else {
+                    if ( !versionMetadata.getDist().getTarball().endsWith(packagePath.getTarPath()) )
```

Comment:
There is a potential NULL pointer exception if the getTarball() returning NULL.

Reply:
I thought tarball can't be null if dist is not
fixed